firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/ApkUpdater.java

```diff
@@ -41,35 +41,40 @@
   private static final int UPDATE_INTERVAL_MS = 250;
   private static final String TAG = "UpdateApkClient:";
   private static final String REQUEST_METHOD = "GET";
-  private final FirebaseAppDistributionNotificationsManager appDistributionNotificationsManager;
 
   private TaskCompletionSource<File> downloadTaskCompletionSource;
   private final Executor taskExecutor; // Executor to run task listeners on a background thread
   private final FirebaseApp firebaseApp;
   private final ApkInstaller apkInstaller;
+  private final FirebaseAppDistributionNotificationsManager appDistributionNotificationsManager;
 
   @GuardedBy("updateTaskLock")
   private UpdateTaskImpl cachedUpdateTask;
 
   private final Object updateTaskLock = new Object();
 
   public ApkUpdater(@NonNull FirebaseApp firebaseApp, @NonNull ApkInstaller apkInstaller) {
-    this(Executors.newSingleThreadExecutor(), firebaseApp, apkInstaller);
+    this(
+        Executors.newSingleThreadExecutor(),
+        firebaseApp,
+        apkInstaller,
+        new FirebaseAppDistributionNotificationsManager(firebaseApp.getApplicationContext()));
   }
 
+  @VisibleForTesting
   public ApkUpdater(
       @NonNull Executor taskExecutor,
       @NonNull FirebaseApp firebaseApp,
-      @NonNull ApkInstaller apkInstaller) {
-    this.appDistributionNotificationsManager =
-        new FirebaseAppDistributionNotificationsManager(firebaseApp);
+      @NonNull ApkInstaller apkInstaller,
+      @NonNull FirebaseAppDistributionNotificationsManager appDistributionNotificationsManager) {
+    this.appDistributionNotificationsManager = appDistributionNotificationsManager;
```

Comment:
Curious as to the constructor change. And setting the notification manager should follow the order of the constructor and go at the end.